environs/imagedownloads/simplestreams.go

```diff
@@ -120,7 +120,7 @@ func validateArgs(arch, release, stream, ftype string) error {
 	}
 
 	validVersion := false
-	workloadVersions, err := corebase.AllWorkloadVersions(release, stream)
+	workloadVersions, err := corebase.AllWorkloadVersions(corebase.Base{}, stream)
```

Comment:
NOTE: Here `release` was included by mistake. `release` here was never a valid series, does not do anything

Reply:
Something is wrong here, none of the calls to AllWordloadVersions include a first argument. Either the signature of the function needs to change, or we're not using it correctly.

Reply:
It is quite odd, the signature to these methods. The 'requestedSeries' param in particular never made much sense to me.

Both `requestedSeries` and `ImageStream` params are [only used in 1 place](https://github.com/juju/juju/blob/3.6/core/base/supportedseries.go#L86-L88), to set the requested series to supported when calling `ControllerSeries`. `WorkloadSeires`, `AllWorkloadVersions`, `AllWorkloadOSTypes`.

I struggled to make much sense of this through my work on this code, so as a result, I ignored it an kept the behaviour identical.

Even before this change, all the calls to `AllWorkloadOSTypes` passed in empty string for `requestedSeries`

I suspect this code is a hangover from a hack a long time ago, which _probably_ isn't required anymore, which no one has been brave enough to clean up.

I would be very happy to remove this, but I wanted to minimise unrelated risky changes like this, when this PR already messes with some core juju behaviour

Reply:
This is a ~~hack~~ workaround for if you set your image stream to daily and the series/base is used. Then it will be defaulted to supported. This probably shouldn't be done here, but in the level above which calls the function. Then add a massive comment describing the expected workflow.